packages/truffle-codec-utils/src/types/errors.ts

```diff
@@ -23,10 +23,10 @@ export namespace Errors {
   //For when we need to throw an error, here's a wrapper class that extends Error.
   //Apologies about the confusing name, but I wanted something that would make
   //sense should it not be caught and thus accidentally exposed to the outside.
-  export class DecodingError extends Error {
-    error: GenericError;
-    constructor(error: GenericError) {
-      super(error.message());
+  export class DecodingError extends Error{
+    error: ErrorForThrowing;
```

Comment:
how bout name: `ThrowableError`?

although Harry, I gotta say, I like your naming style :)

Reply:
Well, so `ThrowableError` was my first thought, but that just sounds like the builtin `Error` class...